lib/active_interaction/base.rb

```diff
@@ -214,13 +215,35 @@ def inputs
     #   Example.run!(x: nil)  # => true
     #   Example.run!(x: rand) # => true
     #
+    # @example Nested checks
+    #   class Example < ActiveInteraction::Base
+    #     hash :x, default: {} do
+    #       integer :y, default: nil
+    #     end
+    #     def execute; given?(:x, :y) end
+    #   end
+    #   Example.run!()               # => false
+    #   Example.run!(x: nil)         # => false
+    #   Example.run!(x: {})          # => false
+    #   Example.run!(x: { y: rand }) # => true
+    #
     # @param input [#to_sym]
     #
     # @return [Boolean]
     #
     # @since 2.1.0
-    def given?(input)
-      @_interaction_keys.include?(input.to_sym)
+    def given?(input, *rest)
+      filter_level = self.class
+      input_level = @_interaction_inputs
+
+      [input, *rest].map(&:to_sym).each do |key|
+        filter_level = filter_level.filters[key]
+        if !filter_level || input_level.nil? || !input_level.key?(key)
```

Comment:
I guess this could happen right now with a top level input. I feel like this method is only responsible for telling you if something was given to you or not. It's not a validation. If you were given the wrong value you were still given something. For your original example, `given?(:a, :b, :c)` I think we should return `false`.